pkg/prometheus/statefulset_test.go

```diff
@@ -855,13 +855,13 @@ func TestPrometheusDefaultBaseImageFlag(t *testing.T) {
 			Labels:      labels,
 			Annotations: annotations,
 		},
-	}, prometheusBaseImageConfig, nil, "", 0, nil)
+	}, operatorConfig, nil, "", 0, nil)
 
 	require.NoError(t, err)
 
 	image := sset.Spec.Template.Spec.Containers[0].Image
 	expected := "nondefaultuseflag/quay.io/prometheus/prometheus"
```

Comment:
In pkg/alertmanager/statefulset_test.go, we append `:<default version>` to the image path and check for equality. From my POV, it's a good approach (we want the version number to be present if not specified in the image field).